homeassistant/core.py

```diff
@@ -565,8 +561,8 @@ def async_run_job(
         if asyncio.iscoroutine(target):
             return self.async_create_task(target)
 
-        target = cast(Callable[..., Union[Coroutine[Any, Any, _R], _R]], target)
-        return self.async_run_hass_job(HassJob(target), *args)
+        # cast removed here because it is almost expensive as call_soon
```

Comment:
Is this cheaper / better?

```
target: Callable[..., Union[Coroutine[Any, Any, _R], _R]] = target
```

or wrap it in `if TYPE_CHECKING:`?

Reply:
`target: Callable[..., Union[Coroutine[Any, Any, _R], _R]] = target` still construct the var.

Will test with `if TYPE_CHECKING:`

Reply:
Great suggestion @frenck 

Overhead is gone with the guards and we get to keep the types

<img width="1225" alt="Screen Shot 2022-05-16 at 13 19 18" src="https://user-images.githubusercontent.com/663432/168657273-95faa653-d468-4c09-a4fa-25181aae8772.png">